variables.tf

```diff
@@ -0,0 +1,247 @@
+}
+
+variable "load_balancer_private_whitelisted_ips" {
+  type        = list(string)
+  description = "A list of IP ranges to whitelist for private access."
+  default     = []
+}
+
+variable "load_balancer_autoscaling_group_names" {
+  type        = list(string)
+  description = "The name of the autoscaling groups containing the target instances, so that instances can be attached to the load balancer's target group."
+}
+
+variable "dns_hosted_zone_id" {
+  type        = string
+  description = "The ID of the hosted zone in Route53, under which the DNS record should be created."
+}
+
+variable "dns_records_private" {
+  type        = map(string)
+  description = "A map with dns records to add in dns_managed_zone for private endpoints set as value. Full domain names will be exported in a map for the given key."
+  default     = {}
+}
+
+variable "dns_records_public" {
+  type        = map(string)
+  description = "A map with dns records to add in dns_managed_zone for public endpoints set as value. Full domain names will be exported in a map for the given key."
+  default     = {}
+}
+
+variable "ssl_certificate_subdomain" {
+  type        = string
+  description = "The subdomain name that will be written in the TLS certificate. Can include a wildcard. The hosted zone name will be appended to form the complete domain name."
+}
+
+variable "tags" {
+  type        = map
+  description = "The tags (a map of key/value pairs) to be applied to created resources."
+  default     = {}
+}
+
+/** Advanced Load balancer params */
+
+/* Load balancer params */
+
+variable "idle_timeout" {
+  type        = number
+  description = "The time in seconds that the connection is allowed to be idle. Only valid for Load Balancers of type application."
+  default     = 60
+}
+
+/* Listener params */
+
+variable "ssl_policy" {
+  type        = string
+  description = "The name of the SSL Policy for the listener. Required if protocol is HTTPS or TLS."
+  default     = "ELBSecurityPolicy-2016-08"
+}
+
+/* Target params */
+
+variable "deregistration_delay" {
+  type        = number
+  description = "The amount time for Elastic Load Balancing to wait before changing the state of a deregistering target from draining to unused. The range is 0-3600 seconds. The default value is 300 seconds."
+  default     = 300
+}
+
+variable "slow_start" {
+  type        = number
+  description = "The amount time for targets to warm up before the load balancer sends them a full share of requests. The range is 30-900 seconds or 0 to disable. The default value is 0 seconds."
+  default     = 0
+}
+
+variable "load_balancing_algorithm_type" {
+  type        = string
+  description = "Determines how the load balancer selects targets when routing requests. Only applicable for Application Load Balancer Target Groups. The value is round_robin or least_outstanding_requests. The default is round_robin."
+  default     = "round_robin"
+}
+
+/* Stickiness */
+
+variable "stickiness_type" {
```

Comment:
I think it would be better to be able to specify this attribute individually for public / private load balancers